s3pd/__init__.py

```diff
@@ -103,6 +104,38 @@ def download_chunk(bucket, key, shmfileno, offset_first, offset_last, signed):
             Range='bytes=%s-%s' % (offset_first, offset_last))['Body']
         chunk._raw_stream.readinto(shmmap)
 
+def resolve_link(bucket, key, signed, depth=10):
+    # Stop after too many link indirections
+    assert depth > 0, 'Too many levels of link indirections'
+
+    client = create_client(signed)
+    filesize = get_filesize(client, bucket, key)
+
+    # There is no need to resolve files with a size >1KB, these could not
+    # realistically be links
+    if filesize > 1024:
+        return bucket, key
+
+    link_sentinel = '#S3PDLINK#'
+    with BytesIO() as stream:
+        client.download_fileobj(Bucket=bucket, Key=key, Fileobj=stream)
+        content = stream.getvalue().decode('utf-8').strip()
+        # Check whether this file is a link
+        if content.startswith(link_sentinel):
```

Comment:
Early return instead ?

```suggestion
        if not content.startswith(link_sentinel):
            return bucket, key
```

Reply:
Fixed in 8d147ce